### PR TITLE
Fix all clippy lint issues and run rustfmt

### PR DESCRIPTION
**Pull request by zmanian**

### CODE REVIEW

**Comment by zmanian** [comment]
Clippy command.

`rustup run nightly-2016-07-20 cargo install clippy`

`rustup run nightly-2016-07-20 cargo clippy`

**Comment by mcginty** [comment]
wanna re-run this on HEAD? I just pushed some fixes to unbreak master.

**Comment by zmanian** [comment]
Yeah I'll do that

**Comment by zmanian** [comment]
Reran against HEAD

File: src/lib.rs [hunk 1]
```diff
@@ -280,41 +296,41 @@ pub fn download(remote_ssh_host : &str,
...
-            Ok(msg) => {
-                if msg.len() == 0 {
-                    die!("failed to get filesize from server, probable timeout.");
+
+        if let Ok(msg) = conn.recv() {
+            if msg.is_empty() {
+                die!("failed to get filesize from server, probable timeout.");
+            }
+            let mut rdr = Cursor::new(msg);
+            filesize = filesize.or_else(|| Some(rdr.read_u64::<LittleEndian>().unwrap()));
```
Comment (mcginty) [inline src/lib.rs]: don't understand why this is chosen over or()

Comment (zmanian) [inline src/lib.rs]: https://github.com/Manishearth/rust-clippy/wiki#or_fun_call

File: src/lib.rs [hunk 2]
```diff
@@ -238,25 +254,24 @@ impl<'a> Server<'a> {
...
 
-pub fn download(remote_ssh_host : &str,
-                port_range      : connection::PortRange,
-                remote_path     : &str,
-                local_path      : PathBuf)
-{
+pub fn download(remote_ssh_host: &str,
+                port_range: connection::PortRange,
+                remote_path: &str,
+                local_path: PathBuf) {
```
Comment (mcginty) [inline src/lib.rs]: i need to find a way to tell rustfmt to align more like the previous

moxie has ruined me with this now any other way looks weird :smile:

but yeah, let's start with a basepoint of rustfmt-ted code and then i can whine about style and try to play with the fmt config later.

Comment (zmanian) [inline src/lib.rs]: I <3 golangs I won't compile until you format the Rob Pike way.

Rust wanted to do that and just didn't have time before 1.0.

**Pull request closed without merging**